apache/superset
# Docker container building and first run improvements

### CATEGORY

Choose one


### SUMMARY
When running a Docker version of superset, the previously recommended way to initialize the db was to exec into the running container _locally_ while executing against production configuration and then manually executing the `flask fab admin-create` command. This was not a sustainable pattern and I wanted to create a better solution that behaved more like a CI deployment. 
* This adds some python scripts that will execute the admin command
* It looks for a secrets provider and falls back to env variables
* Provides some examples of env setup and superset_config.py

### TEST PLAN
* Setup a bare db to be your new environment (this usually means already having created users/schemas etc.)
* Configure your docker.env file to have the connection details for this db
* Build the container
* You can run it locally (just to verify) or you can deploy it out to ECS or similar
* The expected results are that all steps to initialize superset are executed in a hands off manner and you can log in when all is said and done.

### ADDITIONAL INFORMATION



### REVIEWERS
